presto-raptor/src/main/java/com/facebook/presto/raptor/systemtables/ColumnRangesSystemTable.java

```diff
@@ -0,0 +1,187 @@
+    {
+        String metadataSqlQuery = getColumnRangesMetadataSqlQuery(sourceTable, indexedRaptorColumns);
+        List<Type> columnTypes = tableMetadata.getColumns().stream()
+                .map(ColumnMetadata::getType)
+                .collect(toImmutableList());
+
+        ResultSetValues resultSetValues = new ResultSetValues(columnTypes);
+
+        try (Connection connection = dbi.open().getConnection();
+                Statement statement = connection.createStatement();
+                ResultSet resultSet = statement.executeQuery(metadataSqlQuery)) {
+            if (resultSet.next()) {
+                resultSetValues.extractValues(resultSet, ImmutableSet.of(), ImmutableSet.of());
+                List<Object> record = IntStream.range(0, columnTypes.size())
+                        .boxed()
+                        .map(field -> getColumnValue(resultSetValues, field, columnTypes.get(field)))
+                        .collect(toList());
+                return new InMemoryRecordSet(columnTypes, ImmutableList.of(record)).cursor();
+            }
+            else {
+                return new InMemoryRecordSet(columnTypes, ImmutableList.of()).cursor();
+            }
+        }
+        catch (SQLException | DBIException e) {
+            throw metadataError(e);
+        }
+    }
+
+    private Object getColumnValue(ResultSetValues resultSetValues, int field, Type columnType)
+    {
+        Class<?> javaType = columnType.getJavaType();
+        if (resultSetValues.isNull(field)) {
+            return null;
+        }
+
+        if (javaType == double.class) {
+            return resultSetValues.getDouble(field);
+        }
+        else if (javaType == long.class) {
+            return resultSetValues.getLong(field);
+        }
+        else if (javaType == boolean.class) {
+            return resultSetValues.getBoolean(field);
+        }
+        else if (javaType == Slice.class) {
+            return resultSetValues.getSlice(field);
+        }
+        else {
+            throw new VerifyException("Unknown Java type: " + javaType);
+        }
+    }
+
+    private static boolean isIndexedType(Type type)
+    {
+        // We only consider the following types in the column_ranges system table
+        // Exclude INTEGER because we don't collect column stats for INTEGER type.
+        // Exclude DOUBLE because Java double is not completely compatible with MySQL double
+        // Exclude VARCHAR because they can be truncated
+        return type.equals(BOOLEAN) || type.equals(BIGINT) || type.equals(DATE) || type.equals(TIMESTAMP);
+    }
+
+    private static String getColumnRangesMetadataSqlQuery(RaptorTableHandle raptorTableHandle, List<TableColumn> raptorColumns)
+    {
+        String shardIndexTableName = shardIndexTable(raptorTableHandle.getTableId());
+        List<String> aggregatedColumns = raptorColumns.stream()
```

Comment:
rename to simpler variable name (columns).

Reply:
Also, maybe consider add separate temporal column since it will most likely be queried by user?